core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala

```diff
@@ -339,7 +339,8 @@ class HadoopMapReduceWriteConfigUtil[K, V: ClassTag](conf: SerializableConfigura
 
   override def createCommitter(jobId: Int): HadoopMapReduceCommitProtocol = {
     FileCommitProtocol.instantiate(
-      className = classOf[HadoopMapReduceCommitProtocol].getName,
+      className = getConf.get("mapreduce.sources.commitProtocolClass",
```

Comment:
The problem here is that we're super conservative in RDD and core API now, and won't likely add something new except bug fixes.

Reply:
> The problem here is that we're super conservative in RDD and core API now, and won't likely add something new except bug fixes.

Okay. I understand about it. Thank you for your comment.  I will try to your suggested way. Fortunately, I am using scala.